src/synapse.py

```diff
@@ -57,13 +59,20 @@ def synapse_environment(self) -> typing.Dict[str, str]:
         Returns:
             A dictionary representing the Synapse environment variables.
         """
-        return {
+        environment = {
             "SYNAPSE_SERVER_NAME": f"{self._charm_state.server_name}",
             "SYNAPSE_REPORT_STATS": f"{self._charm_state.report_stats}",
             # TLS disabled so the listener is HTTP. HTTPS will be handled by Traefik.
             # TODO verify support to HTTPS backend before changing this  # pylint: disable=fixme
             "SYNAPSE_NO_TLS": str(True),
         }
+        if self._database_data is not None:
+            environment["POSTGRES_DB"] = self._database_data.get("POSTGRES_DB", "")
```

Comment:
Would this work? https://github.com/canonical/synapse-operator/pull/13/files#r1253285244

Reply:
It seems that we have many cases where the point is what to do if something from a relation is required but is not available. Should block the charm?

Reply:
yeh probably blocking the charm or error state since the operator probably can't do much at that point?

Reply:
Exactly. I'm considering leaving it as ErrorStatus because the relation with Postgresql is not required and maybe the BlockedStatus would give this impression.